File: poh4/poh4_3.go
//usr/bin/env go run $0 $@;exit
package main
import(
	"fmt"
	"os"
	"text/scanner"
	"strconv"
)

var sin scanner.Scanner
func scanint() int{
	sin.Scan()
	ret,_ := strconv.Atoi(sin.TokenText())
	return ret
}

func main(){
	sin.Init(os.Stdin)
	T:=scanint()
	N:=scanint()
	a:=make([]int,N)
	cur:=0

	for i:=0;i<T;i++ {
		a[i]=scanint()
		cur+=a[i]
	}
	r:=cur
	for i:=T;i<N;i++ {
		a[i]=scanint()
		cur+=a[i]
		cur-=a[i-T]
		if r<cur {r=cur}
	}
	fmt.Println(r)
}

File: poh4/poh4_3.R
#!/usr/bin/env Rscript
R=scan("stdin")
T=R[1]
N=R[2]
cur=i=0
while(i<T){
	cur=cur+R[i+3]
	i=i+1
}
r=cur
while(i<N){
	cur=cur+R[i+3]-R[i+3-T]
	if(r<cur)r=cur
	i=i+1
}
cat(r)